source/ConditionSet.cpp

```diff
@@ -364,17 +364,17 @@ void ConditionSet::Add(const DataNode &node)
 bool ConditionSet::Add(const string &firstToken, const string &secondToken)
 {
 	// Each "unary" operator can be mapped to an equivalent binary expression.
-	if(firstToken == "not")
+	if(firstToken == "not" && DataNode::IsConditionName(secondToken))
```

Comment:
I was under the impression that the C++ stdlib regex is pretty slow, should we calculate these at the start of the function and reuse them? Might be too much of a premature optimisation though.

Reply:
IsConditionName will only be executed if the first part of the expression (firstToken == "not") is true. Otherwise c++ will immediately move on to the else section.

Reply:
Might be problematic if someone is writing expressions like `not ++`, because that will be accepted on line 375 (with not being a condition name). So IsConditionName should also filter out existing keywords to properly cover this.

Reply:
I'll keep this for a follow-up PR. (This PR doesn't make the current situation any worse.)